android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -0,0 +1,257 @@
+        searchRecyclerView.setAdapter(globalSearchAdapter);
+        searchRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        if (savedInstanceState != null && savedInstanceState.getString(SEARCH) != null) {
+            searchText = savedInstanceState.getString(SEARCH);
+        }
+
+        if (Intent.ACTION_SEARCH.equals(getIntent().getAction())) {
+            handleIntent(getIntent());
+        }
+
+    }
+
+    @Override
+    protected int getLayoutResource() {
+        return R.layout.activity_search;
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_home, menu);
+        SearchManager searchManager =
+                (SearchManager) getSystemService(Context.SEARCH_SERVICE);
+        searchView = (SearchView) menu.findItem(R.id.action_search_home).getActionView();
+        searchView.setSearchableInfo(
+                searchManager.getSearchableInfo(getComponentName()));
+
+        searchView.setOnQueryTextListener(this);
+        if (searchText != null) {
+            searchView.setQuery(searchText,true);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query) {
+        searchQuery(query);
+        searchText = query;
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        searchView.clearFocus();
+        return true;
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        handleIntent(intent);
+    }
+
+    public void handleIntent(Intent intent){
+        final String query = intent.getStringExtra(SearchManager.QUERY);
+        searchQuery(query);
+    }
+
+    private void searchQuery(String constraint) {
+
+        results.clear();
+
+        if (!TextUtils.isEmpty(constraint)) {
+            String query = constraint.toLowerCase(Locale.getDefault());
+            Boolean headerSet = false;
+            //Tracks Filtered Here
+            List<Track> trackList = dbSingleton.getTrackList();
+            int trackResultCount = 0;
+            for (Track track : trackList) {
+                final String text = track.getName().toLowerCase(Locale.getDefault());
+                if (text.contains(query)) {
+                    if(!headerSet) {
+                        results.add("Tracks");
+                        headerSet = true;
+                    }
+                    results.add(track);
+                    trackResultCount++;
+                }
+            }
+            globalSearchAdapter.notifyDataSetChanged();
+            Timber.d("Filtering of tracks done total results %d", trackResultCount);
+
+            //Speakers Filtered here
+            int speakerResultCount = 0;
+            headerSet = false;
+            List<Speaker> speakerList = dbSingleton.getSpeakerList(DbContract.Speakers.NAME);
```

Comment:
@iamareebjamal I am facing an issue after doing the refactor. The order in which the location results and the speaker results appear aren't defined now. Sometimes speakers come after locations ans sometimes its vice versa. Am i doing something wrong here or is it supposed to work like that?